Paws4Me/Paws4Me/Controllers/Logic/ViewController.swift

```diff
@@ -8,109 +8,227 @@
 import UIKit
 
 class ViewController: UIViewController, UISearchBarDelegate {
+    // MARK: - IBOulets
     @IBOutlet weak private var petTable: UITableView!
     @IBOutlet weak private var searchBar: UISearchBar!
-    private var filteredPetObject: AdoptPet?
-    private var adoptPetObject: AdoptPet?
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        getAdoption()
-        petTable.delegate = self
-        petTable.dataSource = self
-        self.title = "Adoptable Animals"
-        setUpSearchbar()
-    }
+    @IBAction func btnCatTapped(_ sender: UIButton) {
+           petType = sender.titleLabel?.text?.lowercased() ?? ""
+           sender.layer.borderColor = UIColor.myAppPurple.cgColor
+           sender.layer.borderWidth = 2
+           searchPetType(type: petType)
+       }
+       @IBAction func btnKittenTapped(_ sender: UIButton) {
+           petType = sender.titleLabel?.text?.lowercased() ?? ""
+           sender.layer.borderColor = UIColor.myAppPurple.cgColor
+           sender.layer.borderWidth = 2
+           searchPetType(type: petType)
+       }
+       @IBAction func btnDogTapped(_ sender: UIButton) {
+           petType = sender.titleLabel?.text?.lowercased() ?? ""
+           sender.layer.borderColor = UIColor.myAppPurple.cgColor
+           sender.layer.borderWidth = 2
+           searchPetType(type: petType)
+       }
+       @IBAction func btnPuppyTapped(_ sender: UIButton) {
+           petType = sender.titleLabel?.text?.lowercased() ?? ""
+           sender.layer.borderColor = UIColor.myAppPurple.cgColor
+           sender.layer.borderWidth = 2
+           searchPetType(type: petType)
+       }
+    // MARK: - Vars/Lets
+       var searchBarController = UISearchBar()
+       var filteredPetObject: AdoptPet?
+       var adoptPetObject: AdoptPet?
+       var isSingleSearch = false
+       var indexSinglePet: Int?
+       var selectedGender: String = ""
+       var petType = ""
+    // MARK: - Life cycle
+       override func viewDidLoad() {
+           super.viewDidLoad()
+           getAdoption()
+           petTable.delegate = self
+           petTable.dataSource = self
+           self.title = "Adoptable Animals"
+           setUpSearchbar()
+       }
+    // MARK: - Functions
     func getAdoption() {
-        guard let url = URL(string: Constants.adoptURL) else { fatalError("Missing URL") }
-        var request = URLRequest(url: url)
-        request.httpMethod = "GET"
-        request.addValue("application/json", forHTTPHeaderField: "Accept")
-        URLSession.shared.makeRequest(url: request as URLRequest, model: AdoptPet.self) { [weak self] result in
+        let urlString = Constants.adoptURL
+        if let request = URL(string: urlString) {
+        URLSession.shared.makeRequest(url: request as URL, model: AdoptPet.self) { [weak self] result in
```

Comment:
So no need to cast as `URL` because you are sending a `URL` 😄 

So this line will change to: 
`URLSession.shared.makeRequest(url: request, model: AdoptPet.self) { [weak self] result in`